homeassistant/components/camera/foscam.py

```diff
@@ -75,20 +75,32 @@ def motion_detection_enabled(self):
 
     def enable_motion_detection(self):
         """Enable motion detection in camera."""
-        ret, err = self._foscam_session.enable_motion_detection()
-        if ret == FOSCAM_COMM_ERROR:
-            _LOGGER.debug("Unable to communicate with Foscam Camera: %s", err)
-            self._motion_status = True
-        else:
+        try:
+            ret, err = self._foscam_session.enable_motion_detection()
+            if ret == FOSCAM_COMM_ERROR:
+                _LOGGER.debug("Unable to communicate "
+                              "with Foscam Camera: %s", err)
+                self._motion_status = True
+            else:
+                self._motion_status = False
+        except:
```

Comment:
Please specify the exceptions that Foscam can raise

Reply:
Without having the broad range of Foscam devices to test against this isn’t really achievable to fix this bug. There are at least two cases noted in the upstream repo.

Reply:
You should be able to see what the Python classes are of the exceptions that get raised

Reply:
Looks like both the issues upstream result in TypeErrors. I'll add just those